### PR TITLE
Make asset cleanup more verbose

### PR DESCRIPTION
**Pull request by Martchus**
* Print exact number of hours (might also prevent potential rounding
  problems)
* Print assigned groups or age
* See https://progress.opensuse.org/issues/41483

### CODE REVIEW

File: lib/OpenQA/Task/Asset/Limit.pm [hunk 1]
```diff
@@ -54,23 +62,25 @@ sub _limit {
...
     my $untracked_assets_storage_duration
-      = $OpenQA::Utils::app->config->{misc_limits}->{untracked_assets_storage_duration};
+      = $OpenQA::Utils::app->config->{misc_limits}->{untracked_assets_storage_duration} * 24;
     my $now = DateTime->now();
     for my $asset (@$assets) {
         $update_sth->execute($asset->{max_job} && $asset->{max_job} >= 0 ? $asset->{max_job} : undef, $asset->{id});
         next if $asset->{fixed} || scalar(keys %{$asset->{groups}}) > 0;
 
-        my $age = $now->delta_days(DateTime::Format::Pg->parse_datetime($asset->{t_created}))->in_units('days');
+        my $age        = $now->delta_days(DateTime::Format::Pg->parse_datetime($asset->{t_created}))->in_units('hours');
```
Comment (coolo) [inline lib/OpenQA/Task/Asset/Limit.pm]: careful if you use delta_days it's still rounded to my understanding

File: lib/OpenQA/Task/Asset/Limit.pm [hunk 2]
```diff
@@ -54,23 +62,25 @@ sub _limit {
...
+"Removing asset $asset_name (not in any group, age ($age hours) exceeds limit ($untracked_assets_storage_duration hours)"
+            );
         }
         else {
-            my $asset_name     = $asset->{name};
-            my $remaining_days = $untracked_assets_storage_duration - $age;
+            my $remaining_hours = $untracked_assets_storage_duration - $age;
             OpenQA::Utils::log_warning(
-                "Asset $asset_name is not in any job group, will delete in $remaining_days days");
+                "Asset $asset_name is not in any job group, will delete in $remaining_hours hours");
```
Comment (coolo) [inline lib/OpenQA/Task/Asset/Limit.pm]: for easier parsing larger values, guess we better stay with remaining_days as a fractional number

Comment (Martchus) [reply lib/OpenQA/Task/Asset/Limit.pm]: ok

**Comment by Martchus** [comment]
I've just read the documentation of the Perl time library we use and played with it in the debugger. It is really ... not so nice. So I just go for the epoch and do the calculation myself. This way we have the accuracy of one second for sure. The output is now in days again but preserving the fractional part.

Note that the [Duration](https://metacpan.org/pod/DateTime::Duration) objects are really just a bunch of integers for year, month, days, hours and so on. So if you do `$duration = DateTime::Duration->new(days => 2);` then `$duration->in_units('hours')` yields 0 and not 48. The method name `in_units` implies a conversion but that's not the case. Btw, [I would implement a duration](https://github.com/Martchus/cpp-utilities/blob/master/chrono/timespan.h) by representing everything in ticks using well named methods which actually do the desired conversions.

**Comment by coolo** [comment]
yeah, I prefer your solution. Better remove Duration from our cpanspec requires - it's evil! :)

And I guess rounding the fractional date for output is feasible. Humans are just bad at interpreting 1.27271 days :)

**Comment by Martchus** [comment]
I would keep the fractional part in the output at least for now. The more info we get to debug this the better. And humans can also easily ignore.

**Comment by coolo** [comment]
I meant to round to 1 or 2 digits, but let's collect data

**Pull request merged**